### ISSUE TITLE
Sorting songs by date does not work as expected

### ISSUE THREAD

**Issue by TL-P** [OxygenCobalt/Auxio#797]
### Describe the Bug/Crash

Tested with [3.5.0-dev](https://github.com/OxygenCobalt/Auxio/releases/tag/v3.5.0-dev).

1. Have an album with songs that have different date tags (and other songs to compare)
2. Sort songs by date

The current behaviour is that all songs in one album are sorted like they have the same date tag, more precisely, the oldest date is picked.

### Describe the intended behavior

All songs should be sorted according to their individual date tag. Since Auxio already displays a date range in the album view (e. g. "1974 - 2023"), this seems to be possible.

### What android version do you use?

Android 13

### What device model do you use?

Xiaomi Mi 9T (PixelOS)

### Provide a sample file

I can provide sample files if needed.

### Bug report

N/A

### Duplicates

- [X] I have checked the [Troubleshooting](https://github.com/OxygenCobalt/Auxio/wiki/Troubleshooting) page.
- [X] I have checked this issue for duplicates.
- [X] I have checked that this issue occurs on the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
Yeah, this is a bit messy since historically "sort by date" has meant "sort by the album's date", and I've been carrying it over ever since. I can change this though while still maintaining album grouping for simpler libraries (As I personally want)

**Comment by TL-P** [comment]
Hey, I know this is a minor issue, but I am using v4.0.4 right now and I think the issue is still present @OxygenCobalt.

**Comment by OxygenCobalt** [comment]
Odd, send sample files @TL-P, just enough to reproduce. Avoid copyrighted stuff if possible

**Comment by TL-P** [comment]
Here you go @OxygenCobalt.

[sample files.zip](https://github.com/user-attachments/files/19411138/sample.files.zip)
